### PR TITLE
feat: Add initial documentation for colour palette

### PR DESCRIPTION
**Pull request by VincentSmedinga**
This shows a simple list of the colours we currently have. No semantic grouping or explanation or anything; these are to be discussed.

### CODE REVIEW

File: storybook/storybook-docs/src/color.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { Meta } from "@storybook/blocks";
+import { ColourPalette } from "./components/ColourPalette";
```
Comment (alimpens) [inline storybook/storybook-docs/src/color.stories.mdx:2]: Did you choose not to use [Storybooks ColorPalette](https://storybook.js.org/docs/api/doc-block-colorpalette) on purpose?

File: storybook/storybook-docs/src/color.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+<Meta title="Docs/Assets/Colour" />
+
+# Colour
+
+The basic colours of Amsterdam are white, red, and black.
+We have 8 additional colours and 3 shades of grey.
+
+<ColourPalette>
+  <ColourPalette.Section>
+    <ColourPalette.Tile name="white" rgb="#ffffff" />
```
Comment (alimpens) [inline storybook/storybook-docs/src/color.stories.mdx:13]: Can we get these values from the design tokens? So we don't have to maintain them in 2 places.

File: storybook/storybook-docs/src/color.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    <ColourPalette.Tile name="red" rgb="#ec0000" />
+    <ColourPalette.Tile name="black" rgb="#000000" />
```
Comment (alimpens) [inline storybook/storybook-docs/src/color.stories.mdx:15]: If we're keeping the custom component: maybe change the prop name from `rgb` to `color`?

File: storybook/storybook-docs/src/color.stories.mdx (hunk 1 above)
Comment (VincentSmedinga) [reply storybook/storybook-docs/src/color.stories.mdx]: Yes, my first attempt used it. Its design doesn’t match ours, we don’t have swatches of shades and tints of one hue, and no names to display on the left.

I’ll be replacing their IconGallery with one of our own as well.

File: storybook/storybook-docs/src/color.stories.mdx (hunk 3 above)
Comment (VincentSmedinga) [reply storybook/storybook-docs/src/color.stories.mdx]: I’ll be making a case to move to [HSL or even fancier models](https://www.smashingmagazine.com/2021/11/guide-modern-css-colors/) when the time is right. But if we do, we should probably still display the RGB notation as well. I wanted to keep the options open. Should we change to `color` until then?

File: storybook/storybook-docs/src/color.stories.mdx (hunk 2 above)
Comment (VincentSmedinga) [reply storybook/storybook-docs/src/color.stories.mdx]: Certainly!

The page even reloads automatically when updating a colour token, pretty amazing.

File: storybook/storybook-docs/src/color.stories.mdx (hunk 3 above)
Comment (alimpens) [reply storybook/storybook-docs/src/color.stories.mdx]: Yeah, I think `color` is a more robust api. We could even allow all types of color notation in the `color` prop, and do any conversion necessary in the component itself... But that's something for later.

Comment (VincentSmedinga) [reply storybook/storybook-docs/src/color.stories.mdx]: Alright, done.

**Review by alimpens — APPROVED** [review]

**Pull request merged**